rpcs3/Emu/RSX/GL/GLOverlays.cpp

```diff
@@ -467,30 +467,69 @@ namespace gl
 			"layout(location=0) in vec2 tc0;\n"
 			"layout(location=0) out vec4 ocol;\n"
 			"\n"
+			"vec2 sbs_single_matrix = vec2(2.0,0.4898f);\n"
+			"vec2 sbs_multi_matrix =  vec2(2.0,1.0);\n"
+			"vec2 ou_single_matrix =  vec2(1.0,0.9796f);\n"
+			"vec2 ou_multi_matrix =   vec2(1.0,2.0);\n"
+			"\n"
 			"uniform float gamma;\n"
 			"uniform int limit_range;\n"
-			"uniform int stereo;\n"
+			"uniform int stereo_display_mode;\n"
 			"uniform int stereo_image_count;\n"
 			"\n"
 			"vec4 read_source()\n"
 			"{\n"
-			"	if (stereo == 0) return texture(fs0, tc0);\n"
+			"	if (stereo_display_mode == 0) return texture(fs0, tc0);\n"
 			"\n"
 			"	vec4 left, right;\n"
-			"	if (stereo_image_count == 2)\n"
+			"	if (stereo_image_count == 1)\n"
+			"	{\n"
+			"		if(stereo_display_mode == 1)\n" // anaglyph
+			"		{\n"
+			"			left = texture(fs0, tc0 * vec2(1.f, 0.4898f));\n"
+			"			right = texture(fs0, (tc0 * vec2(1.f, 0.4898f)) + vec2(0.f, 0.510204f));\n"
+			"			return vec4(left.r, right.g, right.b, 1.f);\n"
+			"		}\n"
+			"		if(stereo_display_mode == 2)\n" // side-by-side
```

Comment:
Declare these as defines just before the matrices. We'll need the following enums:
```C
#define STEREO_MODE_DISABLED 0
#define STEREO_MODE_ANAGLYPH 1
#define STEREO_MODE_SIDE_BY_SIDE 2
#define STEREO_MODE_OVER_UNDER 3
```

Also, replace the if-else chain with a switch-case block. It will be much easier to read.